src/vm/virtualMachine.ts

```diff
@@ -66,10 +68,14 @@ export class VirtualMachine {
 
   /**
    * Execute one step:
+   * - Execute hints at PC
    * - Decode the instruction at PC
    * - Run the instruction
    */
-  step(): void {
+  step(hints?: Hint[]): void {
+    if (hints) {
+      hints.map((hint) => HintProcessor.execute(this, hint));
```

Comment:
I really don't like when `this` is given as first argument of a function, why not then updating the class def of this to have the function as a class method?

Reply:
Will refactor HintProcessor into the VirtualMachine class